beacon-chain/blockchain/core.go

```diff
@@ -137,12 +119,12 @@ func (b *BeaconChain) GenesisBlock() (*types.Block, error) {
 
 // CanonicalHead fetches the latest head stored in persistent storage.
 func (b *BeaconChain) CanonicalHead() (*types.Block, error) {
-	headExists, err := b.db.Has([]byte(canonicalHeadKey))
+	headExists, err := b.db.Has(canonicalHeadLookupKey)
```

Comment:
Do we need to do a Has if we are immediately doing a Get? 
Wouldn't Get return nil or some not found error?

Reply:
yeap a get would return an error if the key does not exist, I guess we can do a get only but instead of the current case where we return a nil block and a nil error when the key does not exist, if we only use a get we would get a nil block and an error returned.